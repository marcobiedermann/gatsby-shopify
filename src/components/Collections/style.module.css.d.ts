declare const styles: {
  readonly collections: string;
  readonly collections__title: string;
};
export = styles;
