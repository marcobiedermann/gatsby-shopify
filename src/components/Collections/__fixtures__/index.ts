import { CollectionsProps } from '..';
import collectionPreview from '../../CollectionPreview/__fixtures__';

const collections: CollectionsProps = {
  collections: [collectionPreview],
};

export default collections;
