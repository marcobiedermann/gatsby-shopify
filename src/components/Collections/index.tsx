import React, { FC } from 'react';
import CollectionPreview, { CollectionPreviewProps } from '../CollectionPreview';
import styles from './style.module.css';

export interface CollectionsProps {
  collections: CollectionPreviewProps[];
}

const Collections: FC<CollectionsProps> = (props) => {
  const { collections } = props;

  return (
    <ul className={styles.collections}>
      <li>
        <div>
          <h1 className={styles.collection__title}>Collections</h1>
        </div>
      </li>
      {collections.map((collection) => (
        <li key={collection.shopifyId}>
          <CollectionPreview {...collection} />
        </li>
      ))}
    </ul>
  );
};

export default Collections;
