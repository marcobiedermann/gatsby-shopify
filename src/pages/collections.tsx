import { graphql, Link, useStaticQuery } from "gatsby"
import React from "react"
import Layout from "../components/Layout"

interface ShopifyCollectionQuery {
  title: string
  shopifyId: string
  description: string
  handle: string
}

interface AllShopifyCollectionQuery {
  allShopifyCollection: {
    edges: {
      node: ShopifyCollectionQuery
    }[]
  }
}

const CollectionsPage = () => {
  const { allShopifyCollection } = useStaticQuery<
    AllShopifyCollectionQuery
  >(graphql`
    {
      allShopifyCollection(sort: { fields: [title] }) {
        edges {
          node {
            title
            shopifyId
            description
            handle
          }
        }
      }
    }
  `)

  return (
    <Layout>
      <h1>Collections</h1>
      <ul>
        {allShopifyCollection.edges.map(edge => (
          <li key={edge.node.shopifyId}>
            <h3>
              <Link to={`/collections/${edge.node.handle}`}>
                {edge.node.title}
              </Link>
            </h3>
            <p>{edge.node.description}</p>
          </li>
        ))}
      </ul>
    </Layout>
  )
}

export default CollectionsPage
