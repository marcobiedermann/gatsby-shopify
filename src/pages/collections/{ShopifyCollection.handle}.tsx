import { graphql, PageProps } from 'gatsby';
import React, { FC } from 'react';
import Collection from '../../components/Collection';
import Grid from '../../components/Grid';

interface Source {
  srcSet: string;
  sizes: string;
  type: string;
}

type Layout = 'fixed' | 'fullWidth' | 'constrained';

interface FeaturedImage {
  altText: string;
  gatsbyImageData: {
    images: {
      sources: Source[];
      fallback: {
        src: string;
        srcSet: string;
        sizes: string;
      };
    };
    layout: Layout;
    width: number;
    height: number;
  };
}

interface PriceRangeV2 {
  minVariantPrice: VariantPrice;
}

interface VariantPrice {
  amount: string;
  currencyCode: string;
}

interface Product {
  handle: string;
  featuredImage: FeaturedImage;
  priceRangeV2: PriceRangeV2;
  shopifyId: string;
  title: string;
}

interface DataProps {
  shopifyCollection: {
    description: string;
    products: Product[];
    shopifyId: string;
    title: string;
  };
}

const CollectionPage: FC<PageProps<DataProps>> = (props) => {
  const {
    data: { shopifyCollection },
    location,
  } = props;

  return (
    <>
      <Grid isFull>
        <Collection {...shopifyCollection} />
      </Grid>
    </>
  );
};

export const query = graphql`
  query ($id: String!) {
    shopifyCollection(id: { eq: $id }) {
      description
      products {
        handle
        featuredImage {
          altText
          gatsbyImageData
        }
        priceRangeV2 {
          minVariantPrice {
            amount
            currencyCode
          }
        }
        shopifyId
        title
      }
      shopifyId
      title
    }
  }
`;

export default CollectionPage;
