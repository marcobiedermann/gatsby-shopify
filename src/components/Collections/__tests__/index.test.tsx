import { render } from '@testing-library/react';
import React from 'react';
import Collections from '..';
import collections from '../__fixtures__';

describe('collections component', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<Collections {...collections} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
