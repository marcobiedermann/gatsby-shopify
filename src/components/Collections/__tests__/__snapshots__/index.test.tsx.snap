// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`collections component should render correctly 1`] = `
<ul
  class="collections"
>
  <li>
    <div>
      <h1
        class="collections__title"
      >
        Collections
      </h1>
    </div>
  </li>
  <li>
    <div
      class="collectionPreview"
    >
      <figure>
        <a
          href="/collections/collection"
        >
          <div
            class="gatsby-image-wrapper gatsby-image-wrapper-constrained"
            data-gatsby-image-wrapper=""
            style="position: relative; overflow: hidden; display: inline-block; vertical-align: top;"
          >
            <div
              style="max-width: 635px; display: block;"
            >
              <img
                alt=""
                aria-hidden="true"
                role="presentation"
                src="data:image/svg+xml;charset=utf-8,%3Csvg height='560' width='635' xmlns='http://www.w3.org/2000/svg' version='1.1'%3E%3C/svg%3E"
                style="max-width: 100%; display: block; position: static;"
              />
            </div>
            <div
              aria-hidden="true"
              data-placeholder-image=""
              style="height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
            />
            <picture>
              <source
                data-srcset="https://cdn.shopify.com/s/files/1/1776/8987/products/9f190cba7218c819c81566bca6298c6a_159x140_crop_center.jpg.webp?v=1590345972 159w,https://cdn.shopify.com/s/files/1/1776/8987/products/9f190cba7218c819c81566bca6298c6a_318x280_crop_center.jpg.webp?v=1590345972 318w,https://cdn.shopify.com/s/files/1/1776/8987/products/9f190cba7218c819c81566bca6298c6a_635x560_crop_center.jpg.webp?v=1590345972 635w"
                sizes="(min-width: 635px) 635px, 100vw"
                type="image/webp"
              />
              <img
                alt=""
                data-gatsby-image-ssr=""
                data-main-image=""
                data-src="https://cdn.shopify.com/s/files/1/1776/8987/products/9f190cba7218c819c81566bca6298c6a_635x560_crop_center.jpg?v=1590345972"
                data-srcset="https://cdn.shopify.com/s/files/1/1776/8987/products/9f190cba7218c819c81566bca6298c6a_159x140_crop_center.jpg?v=1590345972 159w,https://cdn.shopify.com/s/files/1/1776/8987/products/9f190cba7218c819c81566bca6298c6a_318x280_crop_center.jpg?v=1590345972 318w,https://cdn.shopify.com/s/files/1/1776/8987/products/9f190cba7218c819c81566bca6298c6a_635x560_crop_center.jpg?v=1590345972 635w"
                decoding="async"
                loading="lazy"
                sizes="(min-width: 635px) 635px, 100vw"
                style="height: 100%; left: 0px; position: absolute; top: 0px; transform: translateZ(0); transition: opacity 250ms linear; width: 100%; will-change: opacity; opacity: 0;"
              />
            </picture>
            <noscript />
            <script
              type="module"
            >
              const t="undefined"!=typeof HTMLImageElement&&"loading"in HTMLImageElement.prototype;if(t){const t=document.querySelectorAll("img[data-main-image]");for(let e of t){e.dataset.src&&(e.setAttribute("src",e.dataset.src),e.removeAttribute("data-src")),e.dataset.srcset&&(e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset"));const t=e.parentNode.querySelectorAll("source[data-srcset]");for(let e of t)e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset");e.complete&&(e.style.opacity=1,e.parentNode.parentNode.querySelector("[data-placeholder-image]").style.opacity=0)}}
            </script>
          </div>
        </a>
      </figure>
      <h3>
        <a
          href="/collections/collection"
        >
          Collection Title
        </a>
      </h3>
    </div>
  </li>
</ul>
`;
